### DISCUSSION TITLE
How to create/update/upload icons to HA?

### DISCUSSION THREAD

**Discussion by GeneralUser-NonDev** [Mariusthvdb/custom-icons#57] [Icon suggestions/requests]
I'm looking to edit (since I can't find) some garage and garage-variant icons (I may do the alert ones too, but not sure yet, depends on how much work it is to do the ones I'm looking for; If I can even figure it out) like the "shutter 0-4" ones, once I figure out how to edit the svg files how do I import it to HA and or send to you to include for others to use? Please explain how in detail as I don't know much of anything with coding and how to do much of anything with this, I have to follow YouTube tutorials to do just about anything on my TrueNAS, Unraid, HA, and Proxmox (just to name the mains). if it matters my HA is a VM on my Proxmox.

Edit: I believe I got the icons edited, Now I just have to figure out how to get them into HA. I'll have to try and find the instructions to send to you for your repo again.

**Comment by GeneralUser-NonDev** [comment]
[HA-Garage-Icons.zip](https://github.com/user-attachments/files/15528670/HA-Garage-Icons.zip)

**Comment by Mariusthvdb** [comment]
hi,
thanksI believe you'd best download the CC, which allows you to drop any .svg icon you like into a folder and use it inside the HA icon pickers.

I wont be adding any more icons to the javascript resource simply because it is way too much hassle, and the CC option is much easier.

If I can find the time, Ill re-organize the repo, but for now suggest you [copy the CC folder](https://github.com/Mariusthvdb/custom-icons/tree/main/custom_components/custom_icons), and give it a try.

please note this is a slimmed down version of the Thomas Loven [Fontawesome repo ](https://github.com/thomasloven/hass-fontawesome)which you could also install to do the same

**Reply by GeneralUser-NonDev** [reply]
> hi, thanksI believe you'd best download the CC, which allows you to drop any .svg icon you like into a folder and use it inside the HA icon pickers.
>
download the what now? Where?

> I wont be adding any more icons to the javascript resource simply because it is way too much hassle, and the CC option is much easier.
> 
> If I can find the time, Ill re-organize the repo, but for now suggest you [copy the CC folder](https://github.com/Mariusthvdb/custom-icons/tree/main/custom_components/custom_icons), and give it a try.
>
would this be through command line?  where/how?

> please note this is a slimmed down version of the Thomas Loven [Fontawesome repo ](https://github.com/thomasloven/hass-fontawesome)which you could also install to do the same

Sorry, as my s/n suggests I don't know coding and such for systems like these, hence why I was looking for and easy "drag-n-drop" way to add to a folder or something and be able to use.

**Reply by Mariusthvdb** [reply]
the link I provided points to the Custom component folder, which I suggested you should copy and install (paste) in your own system.

you can easily do that if you install the Samba add-on in HA, and connect to the instance over the network.

then create a custom_icons folder inside your www folder and restart.

upon restart you can use any .svg icon you drop inside that folder in the HA icon picker, with the prefix `cli:` in step of the core HA `mdi:`,

or in Yaml, like so on this view:
```
- title: Custom icons
  path: custom_icons
  icon: cli:adore
  cards:

```

**Reply by GeneralUser-NonDev** [reply]
Thank you for the run down, I'll give this a shot. Question, can the icons be in sub folders in the "custom_icons" folder, or do they all have to be in the top level/main folder?
